airflow/models/taskinstance.py

```diff
@@ -1751,15 +1749,6 @@ def _handle_reschedule(
     ti.end_date = timezone.utcnow()
     ti.set_duration()
 
-    # Lock DAG run to be sure not to get into a deadlock situation when trying to insert
-    # TaskReschedule which apparently also creates lock on corresponding DagRun entity
-    with_row_locks(
-        session.query(DagRun).filter_by(
-            dag_id=ti.dag_id,
-            run_id=ti.run_id,
-        ),
-        session=session,
-    ).one()
```

Comment:
Is this related? (I’m not saying I understand what the code does to begin with)

Reply:
Yes it is.  By doing `for key share update` that means we no longer block `select for key share` which means this insert (few lines below) will not conflict with dag run locks.   so there's no reason to lock the dag run!

Reply:
removing this lock is really the main point of this PR.  using the weaker lock in scheduler is in service of this.